abstract-bits-derive/src/model.rs

```diff
@@ -187,18 +196,61 @@ impl Field {
     }
 }
 
-fn max_size_from_control_list(ident: &syn::Ident, previous_fields: &[Field]) -> usize {
-    if let Some(bits) = previous_fields.iter().find_map(|f| match f {
-        Field::ControlList {
-            controlled, bits, ..
-        } if controlled == ident => Some(bits),
+fn max_size_from_controller_field(
+    controller_expr: &syn::Expr,
+    previous_fields: &[Field],
+) -> usize {
+    // Extract the base field name from the expression
+    let controller_ident = match controller_expr {
+        syn::Expr::Path(path) if path.path.segments.len() == 1 => {
+            &path.path.segments[0].ident
+        }
+        syn::Expr::Field(field_expr) => {
```

Comment:
body of this match arm might be  good candidate for extract function. There is a lot happening (quite some control flow) and the return type is simple (just the ident as all other paths diverge (lead to abort))